File: webapp/rust/infra/src/tables/livestream_viewers_history.rs
use isupipe_core::models::livestream::LivestreamId;
use isupipe_core::models::user::UserId;

qbey::qbey_schema!(
    LivestreamViewersHistoryTable,
    "livestream_viewers_history",
    [id: i64, user_id: UserId, livestream_id: LivestreamId, created_at: i64], row = LivestreamViewersHistoryRow);

pub const TABLE_LIVESTREAM_VIEWERS_HISTORY: LivestreamViewersHistoryTable =
    LivestreamViewersHistoryTable::new();

File: webapp/go/domain/livestream_viewers_history.go
package domain

type LivestreamViewersHistoryID = ID[LivestreamViewersHistoryModel]

// LivestreamViewersHistoryModel はライブ配信の視聴履歴 (livestream_viewers_history) の1行。
type LivestreamViewersHistoryModel struct {
	ID           LivestreamViewersHistoryID `db:"id"`
	UserID       UserID                     `db:"user_id"`
	LivestreamID LivestreamID               `db:"livestream_id"`
	CreatedAt    int64                      `db:"created_at"`
}
